src/minErr.js

```diff
@@ -34,19 +34,32 @@ function minErr(module, ErrorConstructor) {
   ErrorConstructor = ErrorConstructor || Error;
   return function() {
     var SKIP_INDEXES = 2;
+    var DEFAULT_MAX_DEPTH = 3;
 
     var templateArgs = arguments,
+      argsDebuggedStringIndexes = [],
       code = templateArgs[0],
       message = '[' + (module ? module + ':' : '') + code + '] ',
       template = templateArgs[1],
       paramPrefix, i;
 
-    message += template.replace(/\{\d+\}/g, function(match) {
-      var index = +match.slice(1, -1),
-        shiftedIndex = index + SKIP_INDEXES;
+    function getArgumentAndConvertToDebugString(index, maxDepth) {
```

Comment:
Usually, `anguar.xyz` helpers are assigned to a `xyz` function that is available inside the Angular closure (and globally in tests, because there is no closure). See `copy` for example.

I expect the code to look something like this:

```js
var errorMaxDepth = ...;
function setErrorMaxDepth(newErrorMaxDepth) {
  errorMaxDepth = newErrorMaxDepth;
}

function minErr(...) {
  // Use `errorMaxDepth`...
}
```

##
Now in tests, you have direct access to both `errorMaxDepth` and `setErrorMaxDepth`:

```js
var originalErrorMaxDepth = errorMaxDepth;

afterEach(function() {
  errorMaxDepth = originalErrorMaxDepth;
});

it('...', function() {
  setErrorMaxDepth(...);
  expect(testError(...).message).toMatch(...);
});
```